corehq/apps/users/validation.py

```diff
@@ -11,14 +11,16 @@ def validate_mobile_username(username, domain, is_unique=True):
     Raises a ValidationError if any issue with the complete username is encountered
     :param username: str, expects complete username ('username@example.commcarehq.org')
     :param domain: str, required
+    :param is_unique: should be True when a new user is being created and false when a user is being updated.
     """
     if not username:
         raise ValidationError(_("Username is required."))
 
     _validate_complete_username(username, domain)
 
-    if not is_username_available(username) and is_unique:
-        raise ValidationError(_("Username '{}' is already taken or reserved.").format(username))
+    if is_unique:
+        if not is_username_available(username):
```

Comment:
Python will automatically evaluate the entire expression as false if the first part of an `and` statement is false so no need to do the nested ifs. 
```suggestion
    if is_unique and not is_username_available(username):
```